### PR TITLE
[TAS-374] ✨ Support purchase multiple wNFT by Stripe

### PR DESCRIPTION
**Pull request by WeiJun0827**

### CODE REVIEW

File: src/util/api/likernft/index.ts [hunk 1]
```diff
@@ -32,15 +32,6 @@ export async function getISCNPrefixByClassId(classId) {
   return decodeURIComponent(doc.id);
 }
 
-export async function getISCNPrefixByClassIdFromChain(classId) {
```
Comment (williamchong) [inline src/util/api/likernft/index.ts]: Why remove from chain functions?

File: src/routes/likernft/fiat/stripe.ts [hunk 2]
```diff
@@ -141,81 +114,67 @@ router.get(
...
   async (req, res, next) => {
     try {
-      const classId = req.query.class_id as string;
       const { wallet } = req.query;
       if (!(wallet || LIKER_NFT_PENDING_CLAIM_ADDRESS) && !isValidLikeAddress(wallet)) throw new ValidationError('INVALID_WALLET');
-      const { iscnPrefix } = res.locals;
-      const promises = [getNFTClassDataById(classId)];
       const {
-        nftId = '', seller = '', memo, email,
+        collects, memo, email,
```
Comment (williamchong) [inline src/routes/likernft/fiat/stripe.ts]: Please name it `collectInfoList` or `collectNftInfoList` or similar names

File: src/util/api/likernft/fiat/stripe.ts [hunk 3]
```diff
@@ -276,3 +281,84 @@ export async function processStripeFiatNFTPurchase(session, req) {
...
+    const { nftId, seller } = firstListing;
+    result.nftId = nftId;
+    result.seller = seller;
+  }
+  return result;
+}
+
+export async function getNFTPurchaseInfo({
+  classId,
+  nftId = '',
```
Comment (williamchong) [inline src/util/api/likernft/fiat/stripe.ts]: Is nftId a required param here?

**Review by williamchong — CHANGES_REQUESTED** [review]
Need to also handle `processFiatNFTPurchase`

File: src/util/api/likernft/index.ts (hunk 1 above)
Comment (WeiJun0827) [reply src/util/api/likernft/index.ts]: No function is using it, can still keep it

File: src/util/api/likernft/fiat/stripe.ts (hunk 3 above)
Comment (WeiJun0827) [reply src/util/api/likernft/fiat/stripe.ts]: It's optional, only for listing class

File: src/routes/likernft/fiat/stripe.ts [hunk 4]
```diff
@@ -141,81 +99,63 @@ router.get(
...
+        name = `${name} + ${classMetadataList.length - 1} more`;
+      }
+
       const claimToken = randomBytes(32).toString('base64url');
+
+      const iscnAndClassIdLog = {};
+      classIds.forEach((classId, i) => {
+        iscnAndClassIdLog[`iscnPrefix${i + 1}`] = iscnPrefixes[i];
+        iscnAndClassIdLog[`classId${i + 1}`] = classId;
+      });
```
Comment (williamchong) [inline src/routes/likernft/fiat/stripe.ts:130]: Interesting and smart way to populate stripe metadata

File: src/routes/likernft/fiat/stripe.ts [hunk 5]
```diff
@@ -141,81 +99,63 @@ router.get(
...
+          classId: classIds[0],
           paymentId,
           token: claimToken,
           wallet: wallet as string,
         }),
         client_reference_id: wallet as string || undefined,
         customer_email: email,
-        cancel_url: getLikerLandNFTClassPageURL({ classId }),
+        cancel_url: getLikerLandNFTClassPageURL({ classId: classIds[0] }),
         line_items: [
```
Comment (williamchong) [inline src/routes/likernft/fiat/stripe.ts]: Actually we can list each nft as a single line item, but that would make price calculation very hard in LIKE case

**Review by williamchong — APPROVED** [review]

File: src/routes/likernft/fiat/stripe.ts (hunk 5 above)
Comment (WeiJun0827) [reply src/routes/likernft/fiat/stripe.ts]: Or should we split the `LIKER_NFT_FIAT_FEE_USD` as an additional line item?

**Review by williamchong — APPROVED** [review]

**Review by williamchong — APPROVED** [review]

**Pull request merged**